File: ProjectPos.BackEnd/ProjectPos.Services/CashUpReconDto.cs
using ProjectPos.Services.DTOs;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectPos.Services
{
    public class CashUpReconDto
    {
        public int RecieptNumber { get; set; }
        public string? PaxName { get; set; }
        public string? Name { get; set; }
        public string? Surname { get; set; }
        public decimal? ChangeAmount { get; set; }
        public decimal? PaymentTotal { get; set; }
        public decimal? CashUpTotal { get; set; }
        public decimal? InvoiceValue { get; set; }
        public decimal? CashTotal { get; set; }
        public ICollection<CashReportDto>? CashReport { get; set; }
        public ICollection<CreditReportDto>? CreditReport { get; set; }
        public ICollection<EcoCashReportDto>? EcoCashReport { get; set; }
        public ICollection<CreditCardReportDto>? CreditCardReport { get; set; }
    }
}

File: ProjectPos.BackOffice.FrontEnd/src/proxy/interfaces/cash-up-recon-dto.ts
import { CashReportDto } from "./cash-report-dto";
import { CreditCardReportDto } from "./credit-card-report-dto";
import { CreditReportDto } from "./credit-report-dto";
import { EcoCashReportDto } from "./eco-cash-report-dto";

export interface CashUpReconDto {
    receiptNumber: number;
    paxName?: string;
    name?: string;
    surname?: string;
    changeAmount?: number;
    paymentTotal?: number;
    cashUpTotal?: number;
    invoiceValue?: number;
    cashTotal?: number;
    cashReport?: CashReportDto[] | null;
    creditReport?: CreditReportDto[] | null;
    ecoCashReport?: EcoCashReportDto[] | null;
    creditCardReport?: CreditCardReportDto[] | null;
}
